### ISSUE TITLE
Sizeless array of component example won't check

### ISSUE THREAD

**Issue by arouhan** [OpenModelica/OpenModelica#8222]
I'm testing the penultimate Engine System example provided in the Modelica language Sepcification 3.5 (see penultimate  example in section 9.1 Expandable Connectors). As far as I understand, it is also the same as in MLS 3.6-dev (end of 9.1.3).

I provide a version of the code that checks and runs under Dymola 2021. Checking under OME 1.18.0 the model gives the following error:
`Error: Variable bus.speed[{1, 3}] not found in scope Engine. `

Or did I miss something from this example ?

[TITI.zip](https://github.com/OpenModelica/OpenModelica/files/7602652/TITI.zip)

**Comment by perost** [comment]
The issue is not really with the sizeless array, it's that we don't support referring to variables added to expandable connectors in other contexts than connect-equations. So a smaller test case would be something like:
```Modelica
expandable connector EC
end EC;

connector RealOutput = output Real;

model M
  EC ec;
  RealOutput speed;
equation
  connect(ec.speed, speed);
  ec.speed = 0;
end M;
```
The specification example is the only use of this I've ever seen and the specification doesn't really mention it otherwise, so unless you have a good reason for why this is needed it will be quite a low priority to implement it.

**Comment by casella** [comment]
> The specification example is the only use of this I've ever seen and the specification doesn't really mention it otherwise, so unless you have a good reason for why this is needed it will be quite a low priority to implement it.

I fully agree. Frankly, I don't see the point of accessing the bus connector variable, when you can equally well access the local variables of the connected connector, `sensor.speed` in the cited case.

**Comment by arouhan** [comment]
clear. I would then question the reason that lead to such specification, but that's another story.

**Comment by casella** [comment]
> clear. I would then question the reason that lead to such specification, but that's another story.

No idea, maybe we could ask @HansOlsson.
